### PR TITLE
feat: allow users to load apollo key from file

### PR DESCRIPTION
**Pull request by lleadbet**
Customers sometimes would rather not pass sensitive keys to the router through environment variables out of an abundance of caution related to concerns about leaking them.

This PR allows users to pass a new argument `--apollo-key-path` that takes a file location as an argument, and uses it as the APOLLO_KEY value for use with Uplink and usage reporting.

Addresses a portion of #3264, specifically the APOLLO_KEY.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

This does not fully address #3264 as it doesn't support APOLLO_GRAPH_REF, but addresses the most significant concern of that issue.

**Notes**

I'm unsure where both tests and docs should be updated here; if there's a good place, let me know.

### CODE REVIEW

File: .changesets/feat_feat_key_from_file.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+### feat: allow users to load apollo key from file ([PR #5917](https://github.com/apollographql/router/pull/5917))
+
+Users sometimes would rather not pass sensitive keys to the router through environment variables out of an abundance of caution. To help address this, you can now pass an argument `--apollo-key-path` that takes a file location as an argument which is read and then used as the Apollo key for use with Uplink and usage reporting.
```
Comment (smyrick) [inline .changesets/feat_feat_key_from_file.md:3]:
```suggestion
Users sometimes would rather not pass sensitive keys to the router through environment variables out of an abundance of caution. To help address this, you can now pass an argument `--apollo-key-path` or env var `APOLLO_KEY_PATH`, that takes a file location as an argument which is read and then used as the Apollo key for use with Uplink and usage reporting.
```

**Comment by smyrick** [comment]
Do we also need to make doc changes in the same PR?

**Comment by lleadbet** [comment]
@smyrick - yes, but unsure where in the docs it should live; probably a good question for @apollographql/docs.

**Comment by smyrick** [comment]
@lleadbet What about updating this page for docs? https://github.com/apollographql/router/blob/main/docs/source/configuration/overview.mdx

**Comment by lleadbet** [comment]
> @lleadbet What about updating this page for docs? https://github.com/apollographql/router/blob/main/docs/source/configuration/overview.mdx
> 
> Which is this page on the UI: https://www.apollographql.com/docs/router/configuration/overview/#environment-variables

good shout, updated

**Comment by abernix** [comment]
I'm going to push an empty commit to your PR.

**Comment by abernix** [comment]
Closing this in favor of #5982.

**Pull request closed without merging**